Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -0,0 +1,1114 @@
+                                                  lz(i, j, k) * dy, lz(i, j + 1, k) * dy});
+            amrex::Real Sx_ext = Sx_stab - Sx(i, j, k);
+            int n_borrow = ComputeNBorrowOneFaceExtension(cell, Sx_ext, Sx_mod, flag_info_face_x,
+                                                          flag_ext_face_x, idim);
+
+            borrowing_x_size(i, j, k) = n_borrow;
+            return n_borrow;
+        },
+        [=] AMREX_GPU_DEVICE (int icell, int ps){
+            amrex::Dim3 cell = box.atOffset(icell).dim3();
+            int i = cell.x;
+            int j = cell.y;
+            int k = cell.z;
+            int nborrow = borrowing_x_size(i, j, k);
+            if (nborrow == 0) {
+                borrowing_x_inds_pointer(i, j, k) = nullptr;
+            } else{
+                borrowing_x_inds_pointer(i, j, k) = borrowing_x_inds + ps;
+
+                amrex::Real Sx_stab = 0.5 * std::max({ly(i, j, k) * dz, ly(i, j, k + 1) * dz,
+                                                      lz(i, j, k) * dy, lz(i, j + 1, k) * dy});
+                amrex::Real Sx_ext = Sx_stab - Sx(i, j, k);
+                for (int j_n = -1; j_n < 2; j_n++) {
+                    for (int k_n = -1; k_n < 2; k_n++) {
+                        //This if makes sure that we don't visit the "diagonal neighbours"
+                        if( !(j_n == k_n or j_n == -k_n)){
+                            // Here a face is available if it doesn't need to be extended itself and if its
+                            // area exceeds Sz_ext. Here we need to take into account if the intruded face
+                            // has given away already some area, so we use Sz_red rather than Sz.
+                            // If no face is available we don't do anything and we will need to use the
+                            // multi-face extensions.
+                            if (Sx_mod(i, j + j_n, k + k_n) > Sx_ext
+                                and ( flag_info_face_x(i, j + j_n, k + k_n) == 1
+                                      or flag_info_face_x(i, j + j_n, k + k_n) == 2)
+                                and flag_ext_face_x(i, j, k)) {
+                                Sx_mod(i, j + j_n, k + k_n) -= Sx_ext;
+                                // Insert the index of the face info
+                                *(borrowing_x_inds + ps) = ps;
+                                // Store the information about the intruded face in the dataset of the
+                                // faces which are borrowing area
+                                FaceInfoBox::addConnectedNeighbor(j_n, k_n, ps,
+                                                                 borrowing_x_neigh_faces);
+                                //*(borrowing_x_i_face + ps) = i;
+                                //*(borrowing_x_j_face + ps) = j + j_n;
+                                //*(borrowing_x_k_face + ps) = k + k_n;
```

Comment:
Just double checking if these lines have been commented out on purpose. If so, shall we remove them?

Reply:
Yep! Thanks